src/assets/scripts/airport/Route/RouteModel.js

```diff
@@ -0,0 +1,134 @@
+import _uniqueId from 'lodash/uniqueId';
+
+/**
+ * Symbol that divides each route segment
+ *
+ * @property SEGMENT_SEPERATION_SYMBOL
+ * @type {string}
+ * @final
+ */
+const SEGMENT_SEPERATION_SYMBOL = '.';
+
+/**
+ * @class RouteModel
+ */
+export default class RouteModel {
+    /**
+     * 'BETHL.GRNPA1.KLAS'
+     *
+     * @for RouteModel
+     * @constructor
+     * @param routeString {string}
+     */
+    constructor(routeString) {
+        if (typeof routeString === 'undefined' || typeof routeString !== 'string') {
+            return;
+        }
+
+        if (!this._isValidRouteString(routeString)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Invalid routeString passed to RouteModel. Expected a routeString of the shape ORIGIN.BASE.DESTINATION but instead received ${routeString}`);
+        }
+
+        /**
+         * @property
+         * @type {string}
+         */
+        this._id = _uniqueId();
+
+        /**
+         * @property origin
+         * @type {string}
+         * @default ''
+         */
+        this.origin = '';
+
+        /**
+         * @property base
+         * @type {string}
+         * @default ''
+         */
+        this.base = '';
+
+        /**
+         * @property destination
+         * @type {string}
+         * @default ''
+         */
+        this.destination = '';
+
+        return this._init(routeString);
+    }
+
+    /**
+     * A single string that represents the entire route
+     *
+     * @property routeString
+     * @return {string}
+     */
+    get routeString() {
+        return `${this.origin}.${this.base}.${this.destination}`;
+    }
+
+    /**
+     * Lifecycle method. Should be run only once on instantiation
+     *
+     * @for RouteModel
+     * @method _init
+     * @param routeString {string}
+     * @private
+     */
+    _init(routeString) {
+        const { origin, base, destination } = this._extractSegmentNamesFromRouteString(routeString);
+
+        this.origin = origin;
+        this.base = base;
+        this.destination = destination;
+
+        return this;
+    }
+
+    /**
+     * Destroy this instance
+     *
+     * @for RouteModel
+     * @method destroy
+     */
+    destroy() {
+        this._id = '';
+        this.origin = '';
+        this.base = '';
+        this.destination = '';
+    }
+
+    /**
+     *
+     * @for RouteModel
+     * @method _extractSegmentNamesFromRouteString
+     * @param routeString {string}
+     * @return {object}
+     * @private
+     */
+    _extractSegmentNamesFromRouteString(routeString) {
```

Comment:
Why the underscore prefix for some of these methods? I know that's a thing when calling lodash functions, but that's clearly not what this is. I've also noticed it with various classes' `._id` property. What's the story on the underscore?